internal/dataplane/sendconfig/inmemory_error_handling.go

```diff
@@ -93,6 +93,13 @@ func parseFlatEntityErrors(body []byte, logger logr.Logger) ([]ResourceError, er
 	if err != nil {
 		return resourceErrors, fmt.Errorf("could not unmarshal config error: %w", err)
 	}
+	if len(configError.Flattened) == 0 {
+		if len(configError.Message) > 0 {
+			logger.Error(nil, "config error missing per-resource errors", "message", configError.Message)
+		} else {
+			logger.Error(nil, "config error missing per-resource and message", "message", configError.Message)
```

Comment:
I'm not sure if this is documented anywhere but we have conducted an effort ( e.g. in https://github.com/Kong/kubernetes-ingress-controller/pull/5065) where we've made all logs start from capital letters. I'd suggest we  continue that (ideally linter would guard this).

Secondly, I'd suggest to include an error (can be a named variable) which we'd attach here. `logger.Error(nil, ...)` would imply the `nil` being logged right? Which might be confusing to users.

```suggestion
			logger.Error(nil, "Config error missing per-resource errors", "message", configError.Message)
		} else {
			logger.Error(nil, "Config error missing per-resource and message", "message", configError.Message)
```

Reply:
:+1: as I commented before this logging made an unfortunate design choice IMHO. It will log a message with log level ERROR and field `err=nil` which at first glance looks like someone screwup logging

Reply:
I invoke the ghost of Christmas past in support of the nils: https://github.com/Kong/kubernetes-ingress-controller/pull/4688#discussion_r1354477492